localstack/services/apigateway/templates.py

```diff
@@ -185,7 +185,10 @@ def build_variables_mapping(api_context: ApiInvocationContext):
                 "params": {
                     "path": api_context.path_params,
                     "querystring": api_context.query_params(),
-                    "header": api_context.headers,
+                    # Sometimes we get a werkzeug.datastructures.Headers object, sometimes a dict
+                    # depending on the request. We need to convert to a dict to be able to render
+                    # the template.
+                    "header": dict(api_context.headers),
```

Comment:
I do have a similar change [here](https://github.com/localstack/localstack/pull/6772/files#diff-056962eb9804524e85fe705957e5d78d5e1206e7461cd12db42234f79e51ae54R895). Do you have a reproducible test for this?

Reply:
@calvernaz Dont have time to create one but its easy this is the jist - 

This should fail without this change

```
    def test_api_gateway_sqs_template_integration(self):
        # create target SQS stream
        queue_name = f"queue-{short_uid()}"
        resource_util.create_sqs_queue(queue_name)

        # create API Gateway and connect it to the target queue
        result = connect_api_gateway_to_sqs(
            "test_gateway4",
            stage_name=self.TEST_STAGE_NAME,
            queue_arn=queue_name,
            path=self.API_PATH_DATA_INBOUND,
            template="Action=SendMessage&MessageBody={\"params\":\"$input.params()\", \"body\":$input.body}"
        )

        # generate test data
        test_data = {"spam": "eggs"}
        test_headers = {"X-Test-Header": "test-header-value"}

        url = path_based_url(
            api_id=result["id"],
            stage_name=self.TEST_STAGE_NAME,
            path=self.API_PATH_DATA_INBOUND,
        )
        result = requests.post(url, data=json.dumps(test_data), headers=test_headers)
        assert 200 == result.status_code

        messages = queries.sqs_receive_message(queue_name)["Messages"]
        assert 1 == len(messages)
        message_body = json.loads(messages[0]["Body"])
        assert test_data == json.dumps(message_body["body"])
        assert test_headers == json.loads(message_body["params"].replace("'", '"')).get('header')
        assert test_data == json.loads(messages[0]["Body"])

```

Reply:
You also need this change: 

<img width="856" alt="image" src="https://user-images.githubusercontent.com/16493770/211755727-c004f6fe-44c9-4773-9d60-552d7bfb7d8d.png">

Reply:
Not a problem @amitassaraf, keep them coming 👍